Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -2610,6 +2680,113 @@ namespace Babylon
+            void UpdateInputSourceEventValues(
+                std::set<xr::System::Session::Frame::InputSource::Identifier> selectStarts,
+                std::set<xr::System::Session::Frame::InputSource::Identifier> selectEnds,
+                std::set<xr::System::Session::Frame::InputSource::Identifier> squeezeStarts,
+                std::set<xr::System::Session::Frame::InputSource::Identifier> squeezeEnds,
+                Napi::Env env)
+            {
+                for (const auto& id : selectStarts)
+                {
+                    auto inputSourceFound = m_idToInputSource.find(id);
+                    if (inputSourceFound != m_idToInputSource.end())
+                    {
+                        auto inputSourceVal = inputSourceFound->second.Value();
+                        Napi::Object inputSourceEvent = GenerateXRInputSourceEvent(inputSourceVal, env);
+
+                        for (const auto& [name, callback] : m_eventNamesAndCallbacks)
+                        {
+                            if (name == JS_EVENT_NAME_SELECT_START)
+                            {
+                                callback.Call({inputSourceEvent});
+                            }
+                        }
+
+                        m_activeSelects.insert(id);
+                    }
+                }
+
+                for (const auto& id : squeezeStarts)
+                {
+                    auto inputSourceFound = m_idToInputSource.find(id);
+                    if (inputSourceFound != m_idToInputSource.end())
+                    {
+                        auto inputSourceVal = inputSourceFound->second.Value();
+                        Napi::Object inputSourceEvent = GenerateXRInputSourceEvent(inputSourceVal, env);
+
+                        for (const auto& [name, callback] : m_eventNamesAndCallbacks)
+                        {
+                            if (name == JS_EVENT_NAME_SQUEEZE_START)
+                            {
+                                callback.Call({inputSourceEvent});
+                            }
+                        }
+
+                        m_activeSqueezes.insert(id);
+                    }
+                }
+
+                for (const auto& id : selectEnds)
+                {
+                    auto inputSourceFound = m_idToInputSource.find(id);
+                    if (inputSourceFound != m_idToInputSource.end())
+                    {
+                        auto inputSourceVal = inputSourceFound->second.Value();
+                        Napi::Object inputSourceEvent = GenerateXRInputSourceEvent(inputSourceVal, env);
+
+                        for (const auto& [name, callback] : m_eventNamesAndCallbacks)
+                        {
+                            if (name == JS_EVENT_NAME_SELECT_END)
+                            {
+                                callback.Call({inputSourceEvent});
+                            }
+
+                            if (name == JS_EVENT_NAME_SELECT)
```

Comment:
>if [](start = 28, length = 2)

nit: `name` will never be more than one of these options, so I'd recommend making these `else if` to avoid subsequent string comparisons.

Reply:
I looked at the webXR API again, and since I need to fire select events before I fire selectend, I need to do two separate iterations.